h/static/scripts/controllers/character-limit-controller.js

```diff
@@ -1,29 +1,28 @@
 'use strict';
 
-function CharacterLimitController(element) {
-  var counter = element.parentElement.querySelector(
-    '.js-character-limit-counter');
-  var input = element.querySelector('.js-character-limit-input');
-  var maxlength = parseInt(input.dataset.maxlength);
-
-  counter.textContent = '';
-
-  function updateCounter() {
-    var length = input.value.length;
+var inherits = require('inherits');
 
-    counter.textContent = length + '/' + maxlength;
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
 
-    if (length > maxlength) {
-      counter.classList.add('is-too-long');
-    } else {
-      counter.classList.remove('is-too-long');
-    }
-  }
-
-  updateCounter();
-  input.addEventListener('input', updateCounter);
+function CharacterLimitController(element) {
+  Controller.call(this, element);
 
-  counter.classList.add('is-ready');
+  var self = this;
+  this.refs.characterLimitInput.addEventListener('input', function () {
```

Comment:
You can grep for all uses of `refs` in a particular controller easily enough. An alternative way to address this would be to use a `data-ref` attribute rather than a class name, or just avoid hyphenating the class name.